draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -7972,8 +7991,17 @@ bytes=500-700,601-999
 </t>
 <t>
    If all of the preconditions are true, the server supports the Range header
-   field for the target resource, and the specified range(s) are invalid or
-   unsatisfiable, the server &SHOULD; send a
+   field for the target resource, and the received Range field-value contains
+   an invalid <x:ref>ranges-specifier</x:ref>,
+   the server &SHOULD; send a <x:ref>400 (Bad Request)</x:ref> response.
+</t>
```

Comment:
I'm not entirely happy that we seem to add a nwe normative requirement to produce a 400 here.

Reply:
I don't consider that to be a new requirement. An invalid ranges specifier is something that the specification already disallowed as a bad message, and the standard code response for "client sent a bad message" is 400. The SHOULD is because servers are allowed to ignore Range before parsing it.

Reply:
I do not disagree, but we have a new SHOULD here :-).

Reply:
okay, have a look at https://github.com/httpwg/http-core/pull/1012/commits/f654eff36c7de2dc0d383e303ab3842a4dc9493a